roles/openshift_health_checker/openshift_checks/__init__.py

```diff
@@ -62,6 +69,45 @@ def subclasses(cls):
             for subclass in subclass.subclasses():
                 yield subclass
 
+    def execute_module(self, *args, **kwargs):
+        """Invoke an Ansible module from a check.
+
+        This is the same as invoking _execute_module from an action plugin
+        but task_vars and tmp from the check are used unless specified.
+        So e.g. check.execute_module(module_name="foo", module_args=dict(arg1=...))
+        Return: result hash from module execution.
+        """
+
+        # fill in unspecified task_vars as either positional (3rd) or keyword arg
+        if len(args) > 2 and args[2] is None:
+            args[2] = self.task_vars
+        else:
+            kwargs["task_vars"] = kwargs.get("task_vars", self.task_vars)
+
+        # fill in unspecified tmp as either positional (4th) or keyword arg
+        if len(args) > 3 and args[3] is None:
+            args[3] = self.tmp
+        else:
+            kwargs["tmp"] = kwargs.get("tmp", self.tmp)
```

Comment:
The ansible method signature is rather longer, but I guess it's less hassle to not deal with overlapping args/kwargs and such.

I thought it might be nice to be able to override `tmp` and `task_vars` per-invocation. Don't know if there's a real use case for it though.